### PR TITLE
docs: use component in mdx pages, closes #90

### LINKED ISSUE #90: docs: add usage for angular component in mdx
### Which scope/s are relevant/related to the feature request?

Docs

### Information

Astro has an integration for mdx and I wanted to use an Angular component in the mdx file.
I couldn't find anything about mdx in docs for `@analogjs/astro-angular`.

Here is how it worked for me:

1. You need `@astrojs/mdx` and `@analogjs/astro-angular` installed and configured in `astro.config.mjs`

```js
import { defineConfig } from "astro/config";
import mdx from "@astrojs/mdx";
import angular from "@analogjs/astro-angular";

// https://astro.build/config
export default defineConfig({
  site: "https://example.com",
  integrations: [mdx(), angular()]
});
```

2. Create an Angular component (e.g. `src/components/hello.component.ts`)

```ts
import { NgIf } from "@angular/common";
import { Component, Input } from "@angular/core";

@Component({
  selector: "app-hello",
  standalone: true,
  imports: [NgIf],
  template: `
    <p>Hello from Angular!!</p>

    <p *ngIf="show">{{ helpText }}</p>

    <button (click)="toggle()">Toggle</button>
  `,
})
export class HelloComponent {
  @Input() helpText = "help";

  show = false;

  toggle() {
    this.show = !this.show;
  }
}
```

3. Open/Create a `.mdx` file (e.g. `pages/blog/using-mdx.mdx`) and import the component below the front matter.

```md
---
layout: "../../layouts/BlogPost.astro"
title: "Using MDX"
description: "Lorem ipsum dolor sit amet"
pubDate: "Jul 02 2022"
heroImage: "/placeholder-hero.jpg"
---

import { HelloComponent } from "../../components/hello.component";

<HelloComponent />
```

The Angular component is rendered correctly. To make it interactive add a client directive `client:visible`.

```md
---
layout: "../../layouts/BlogPost.astro"
title: "Using MDX"
description: "Lorem ipsum dolor sit amet"
pubDate: "Jul 02 2022"
heroImage: "/placeholder-hero.jpg"
---

import { HelloComponent } from "../../components/hello.component";

<HelloComponent client:visible />
```

An error occurred after adding the client directive and the component wasn't interactive.

```
TypeError: Failed to fetch dynamically imported module: http://localhost:3000/@fs/Users/m/Dev/wip/astro-analog/src/components/hello.component
```

Solution is to add `ts` suffix to the component import.

```diff
---
layout: "../../layouts/BlogPost.astro"
title: "Using MDX"
description: "Lorem ipsum dolor sit amet"
pubDate: "Jul 02 2022"
heroImage: "/placeholder-hero.jpg"
---

+import { HelloComponent } from "../../components/hello.component.ts";
import { HelloComponent } from "../../components/hello.component";

<HelloComponent client:visible />
```

Now the error is gone, component loaded properly and interactive.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by marcjulian**
* add link to client directives

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-angular-plugin
- [x] astro-angular
- [ ] create-analog

## What is the current behavior?

Issue Number: #90 

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by marcjulian** [comment]
@brandonroberts this PR is ready for review

File: packages/astro-angular/README.md [hunk 1]
```diff
@@ -151,6 +151,54 @@ import { HelloComponent } from '../components/hello.component';
 
 Find more information about [Client Directives](https://docs.astro.build/en/reference/directives-reference/#client-directives) in the Astro documentation.
 
+## Use Component in MDX pages
```
Comment (brandonroberts) [inline packages/astro-angular/README.md:154]:
```suggestion
## Using Components in MDX pages
```

File: packages/astro-angular/README.md [hunk 2]
```diff
@@ -151,6 +151,54 @@ import { HelloComponent } from '../components/hello.component';
...
+
+Install and configure MDX support by following the Astro integration of [@astrojs/mdx](https://docs.astro.build/en/guides/integrations-guide/mdx/). Your `astro.config.mjs` should now include the `@astrojs/mdx` integration.
```
Comment (brandonroberts) [inline packages/astro-angular/README.md:156]:
```suggestion
To use components with MDX pages, you must install and configure MDX support by following the Astro integration of [@astrojs/mdx](https://docs.astro.build/en/guides/integrations-guide/mdx/). Your `astro.config.mjs` should now include the `@astrojs/mdx` integration.
```

File: packages/astro-angular/README.md [hunk 3]
```diff
@@ -151,6 +151,54 @@ import { HelloComponent } from '../components/hello.component';
...
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
+import angular from '@analogjs/astro-angular';
+
+export default defineConfig({
+  integrations: [mdx(), angular()],
+});
+```
+
+Create a `.mdx` file inside the `src/pages` directory and add the Angular component import below the frontmatter.
```
Comment (brandonroberts) [inline packages/astro-angular/README.md:168]:
```suggestion
Create an `.mdx` file inside the `src/pages` directory and add the Angular component import below the frontmatter.
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by marcjulian** [comment]
Thanks for the feedback.

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**